__tests__/status.test.js

```diff
@@ -1,121 +1,105 @@
-import knex from 'knex';
 import { random } from 'faker';
-import config from '../knexfile';
-import app from '../server';
+import { launchApp, shutdownApp, clear } from './base.js';
 
 describe('Status', () => {
   let db;
-  let server;
-  let existingStatus;
+  let app;
 
   beforeAll(async () => {
-    db = knex(config.test);
-    await db.migrate.latest();
-    server = await app();
-  });
-
-  beforeEach(async () => {
-    await server.objection.models.status.query().delete();
-    existingStatus = await server.objection.models.status.query().insert({
-      name: random.word(),
-    });
+    ({ app, db } = await launchApp());
   });
 
   afterAll(async () => {
-    await server.close();
-    await db.destroy();
+    await shutdownApp(app, db);
   });
 
-  describe('read', () => {
-    it('should return 200', async () => {
-      const res = await server.inject({
-        method: 'get',
-        url: '/statuses',
-      });
-      expect(res.statusCode).toBe(200);
-    });
+  beforeEach(async () => {
+    await clear(app);
   });
 
   describe('create', () => {
-    describe('when using valid name', () => {
-      it('should return 302', async () => {
-        const res = await server.inject({
-          method: 'post',
-          url: '/statuses',
-          body: {
-            name: 'name',
-          },
-        });
-        expect(res.statusCode).toBe(302);
+    it('should return 302 when using valid name', async () => {
+      const status = {
+        name: random.word(),
+      };
+      const { statusCode } = await app.inject({
+        method: 'post',
+        url: '/status',
+        body: status,
       });
+      expect(statusCode).toBe(302);
+      const statuses = await app.objection.models.status.query();
```

Comment:
why add comment? its actually pretty simple - since you have ALL the things related to test right here ( in test itself or hook ), you should be pretty sure what state should be there at each point of time! :) that's one of main benefits, you don't need filtering at all!

Reply:
it's not about that. It's about test readability. I look at the test, I see no filtering, I wonder why. Why whould you do that to me =(

Reply:
yeah i get your point, mine is - when you fully control state should you wonder 'why there is no filtering', or 'why there is filtering' ? :) 
since you fully control whole env, why not to use its all advantages to make checks as strict as possible? ( referring to duplicate task problem )

Reply:
Yeah, you are totally right. But test shouldn't be coupled so close to the environment. And future you or me reading this code will be wondering why there is no filtering. Especially if there will be a lot of tests with different environment.  It's good we won't find anything in database. But let's keep it readable on the spot